### PR TITLE
gh-121103: Update site module docs for free-threaded installs

### PR DESCRIPTION
**Pull request by colesbury**
* Issue: gh-121103

----
📚 Documentation preview 📚: https://cpython-previews--122737.org.readthedocs.build/en/122737/library/site.html

### CODE REVIEW

File: Doc/library/site.rst [hunk 1]
```diff
@@ -32,14 +32,19 @@ It starts by constructing up to four directories from a head and a tail part.
 For the head part, it uses ``sys.prefix`` and ``sys.exec_prefix``; empty heads
 are skipped.  For the tail part, it uses the empty string and then
 :file:`lib/site-packages` (on Windows) or
-:file:`lib/python{X.Y}/site-packages` (on Unix and macOS).  For each
+:file:`lib/python{X.Y[t]}/site-packages` (on Unix and macOS), where the
+optional suffix "t" indicates the :term:`free threading` build.  For each
```
Comment (AlexWaygood) [inline Doc/library/site.rst]:
```suggestion
:file:`lib/python{X.Y[t]}/site-packages` (on Unix and macOS). (The
optional suffix "t" indicates the :term:`free threading` build, and is
appended if ``"t"`` is present in the :attr:`sys.abiflags` constant.)
For each
```

File: Doc/library/site.rst [hunk 2]
```diff
@@ -32,14 +32,19 @@ It starts by constructing up to four directories from a head and a tail part.
...
 of the distinct head-tail combinations, it sees if it refers to an existing
 directory, and if so, adds it to ``sys.path`` and also inspects the newly
 added path for configuration files.
 
 .. versionchanged:: 3.5
    Support for the "site-python" directory has been removed.
 
+.. versionchanged:: 3.13
+   The :term:`free threading` Python installations are identified by the "t"
+   suffix in the version-specific directory name, such as :file:`lib/python3.13t/`.
```
Comment (AlexWaygood) [inline Doc/library/site.rst]:
```suggestion
   On Unix, :term:`Free threading <free threading>` Python installations are
   identified by the "t" suffix in the version-specific directory name, such as
   :file:`lib/python3.13t/`.
```

**Review by AlexWaygood — COMMENTED** [review]
Thanks, looks good! Do you also need to amend the `site.USER_BASE` entry?

**Comment by colesbury** [comment]
I think `site.USER_BASE` was not changed.

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**